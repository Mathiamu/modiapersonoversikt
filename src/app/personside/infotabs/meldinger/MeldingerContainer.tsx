import * as React from 'react';
import { Traad } from '../../../../models/meldinger/meldinger';
import styled from 'styled-components';
import theme, { pxToRem } from '../../../../styles/personOversiktTheme';
import RestResourceConsumer from '../../../../rest/consumer/RestResourceConsumer';
import TraadListe from './traadliste/TraadListe';
import { CenteredLazySpinner } from '../../../../components/LazySpinner';
import { useEffect } from 'react';
import { hasData } from '../../../../rest/utils/restResource';
import { huskForrigeValgtTraad } from '../../../../redux/meldinger/actions';
import { useDispatch } from 'react-redux';
import { useAppState, useRestResource } from '../../../../utils/customHooks';
import { erValgtIDyplenke, MeldingerDyplenkeRouteComponentProps, useInfotabsDyplenker } from '../dyplenker';
import { withRouter } from 'react-router';
import TraadVisning from './traadvisning/TraadVisning';
import Verktoylinje from './traadvisning/verktoylinje/Verktoylinje';

const meldingerMediaTreshold = pxToRem(900);

const MeldingerArticleStyle = styled.article`
    @media (min-width: ${meldingerMediaTreshold}) {
        display: flex;
        align-items: flex-start;
        > *:last-child {
            margin-left: ${theme.margin.layout};
        }
    }
    > * {
        margin-bottom: ${theme.margin.layout};
    }
    position: relative;
    > *:first-child {
        flex: 30% 1 1;
    }
    > *:last-child {
        position: sticky;
        top: 0;
        flex: 70% 1 1;
    }
`;

function MeldingerContainer(props: MeldingerDyplenkeRouteComponentProps) {
    const dispatch = useDispatch();
    const forrigeValgteTraad = useAppState(state => state.meldinger.forrigeValgteTraad);
    const traaderResource = useRestResource(resources => resources.tråderOgMeldinger);
    const dyplenker = useInfotabsDyplenker();

    useEffect(() => {
        if (!hasData(traaderResource)) {
            return;
        }
        const traadIUrl = traaderResource.data.find(traad => erValgtIDyplenke.meldinger(traad, props));
        if (!traadIUrl) {
            props.history.push(dyplenker.meldinger.link(forrigeValgteTraad || traaderResource.data[0]));
        } else if (traadIUrl !== forrigeValgteTraad) {
            dispatch(huskForrigeValgtTraad(traadIUrl));
        }
    }, [forrigeValgteTraad, traaderResource, dispatch, props, dyplenker.meldinger]);

    return (
        <RestResourceConsumer<Traad[]>
            getResource={restResources => restResources.tråderOgMeldinger}
            returnOnPending={<CenteredLazySpinner />}
        >
            {data => (
                <MeldingerArticleStyle>
                    <TraadListe traader={data} />
                    <div>
                        <Verktoylinje />
                        <TraadVisning />
                    </div>
                </MeldingerArticleStyle>
            )}
        </RestResourceConsumer>
    );
}

export default withRouter(MeldingerContainer);
