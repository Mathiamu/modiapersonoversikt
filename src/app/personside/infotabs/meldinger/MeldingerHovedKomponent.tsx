import * as React from 'react';
import { Traad } from '../../../../models/meldinger/meldinger';
import TraadListeContainer from './traadliste/TraadListeContainer';

interface VisningProps {
    traader: Traad[];
}

function MeldingerVisning(props: VisningProps) {
    return <TraadListeContainer />;
}

export default MeldingerVisning;
