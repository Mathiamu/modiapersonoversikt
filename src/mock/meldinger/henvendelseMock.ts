import { OpprettHenvendelseResponse } from '../../models/meldinger/meldinger';
import navfaker from 'nav-faker';

export const henvendelseResponseMock: OpprettHenvendelseResponse = {
    behandlingsId: '890',
    oppgaveId: navfaker.random.vektetSjanse(0.5) ? '123' : undefined
};
