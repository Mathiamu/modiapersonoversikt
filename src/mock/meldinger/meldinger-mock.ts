import { LestStatus, Melding, Saksbehandler, Traad, Meldingstype } from '../../models/meldinger/meldinger';
import faker from 'faker/locale/nb_NO';
import navfaker from 'nav-faker';
import dayjs from 'dayjs';
import { fyllRandomListe } from '../utils/mock-utils';
import {
    erMeldingstypeSamtalereferat,
    erVarselMelding,
    erMeldingFraNav,
    saksbehandlerTekst
} from '../../app/personside/infotabs/meldinger/utils/meldingerUtils';
import { Temagruppe, TemaSamtalereferat } from '../../models/temagrupper';
import standardTeksterMock from '../standardTeksterMock';
import { autofullfor, AutofullforMap } from '../../app/personside/dialogpanel/sendMelding/autofullforUtils';
import { backendDatoTidformat } from '../../utils/date-utils';

// Legger inn to konstanter for å sørge for at vi får korrelasjon på tvers av mocking (tråd-oppgave feks)
export const MOCKED_TRAADID_1 = '123';
export const MOCKED_TRAADID_2 = '321';
export const MOCKED_TRAADID_3 = '987';

export function getMockTraader(fødselsnummer: string): Traad[] {
    faker.seed(Number(fødselsnummer));
    navfaker.seed(fødselsnummer + 'meldinger');

    const traadArray = navfaker.random.vektetSjanse(0.2)
        ? fyllRandomListe(getMockTraad, 300, true)
        : fyllRandomListe(getMockTraad, 30, true);

    if (traadArray.length >= 3) {
        traadArray[0].traadId = MOCKED_TRAADID_1;
        traadArray[1].traadId = MOCKED_TRAADID_2;
        traadArray[2].traadId = MOCKED_TRAADID_3;
    }

    return traadArray.map((traad) => {
        const meldinger = traad.meldinger;
        meldinger[0].id = traad.traadId; // Første melding i en tråd har alltid id === traadId
        return {
            ...traad,
            meldinger
        };
    });
}

function getMockTraad(): Traad {
    const temagruppe = navfaker.random.arrayElement([...TemaSamtalereferat, null, Temagruppe.InnholdSlettet]);
    const meldinger = Array(navfaker.random.integer(4, 1))
        .fill(null)
        .map(() => getMelding(temagruppe));

    const enkeltStaaendeMelding = meldinger.find(
        (melding) => erVarselMelding(melding.meldingstype) || erMeldingstypeSamtalereferat(melding.meldingstype)
    );

    return {
        traadId: faker.random.alphaNumeric(8),
        meldinger: enkeltStaaendeMelding ? [enkeltStaaendeMelding] : meldinger
    };
}

function getMelding(temagruppe: Temagruppe): Melding {
    const visKontrosperre = navfaker.random.vektetSjanse(0.1);
    const ferdigstilUtenSvar = navfaker.random.vektetSjanse(0.1);
    const visMarkertSomFeilsendt = navfaker.random.vektetSjanse(0.1);
    const meldingstype = navfaker.random.arrayElement(Object.entries(Meldingstype))[0];
    const sladdingNiva = navfaker.random.arrayElement([0, 0, 0, 0, 1, 1, 1, 1, 2]);

    let tekstFraNav = navfaker.random.arrayElement(
        Object.entries(standardTeksterMock).map((it) => it[1].innhold.nb_NO)
    );

    if (sladdingNiva === 2) {
        tekstFraNav = tekstFraNav.replace(/./g, '*');
    } else if (sladdingNiva === 1) {
        const paragrafer = tekstFraNav.split('\r\n\r\n');
        if (paragrafer.length > 2) {
            paragrafer[0] = sensorerEnkeltOrd(paragrafer[0]);
            paragrafer[1] = paragrafer[1].replace(/\S/g, '*');
            paragrafer[2] = sensorerEnkeltOrd(paragrafer[2]);
        }
        tekstFraNav = paragrafer.join('\r\n\r\n');
    }

    const fritekst = erMeldingFraNav(meldingstype)
        ? autofullfor(tekstFraNav, getMockAutoFullførMap())
        : faker.lorem.sentences(faker.random.number(15));

    return {
        id: faker.random.alphaNumeric(8),
        meldingstype: meldingstype,
        temagruppe: temagruppe,
        skrevetAvTekst: saksbehandlerTekst(getSaksbehandler()),
        journalfortAv: getSaksbehandler(),
        journalfortDato: navfaker.random.vektetSjanse(0.5)
            ? dayjs(faker.date.recent(50)).format(backendDatoTidformat)
            : undefined,
        journalfortSaksid: faker.random.alphaNumeric(5),
        journalfortTemanavn: navfaker.random.arrayElement(['Dagpenger', 'Arbeid', 'Pensjon', 'Bidrag']),
        fritekst: fritekst,
        lestDato: dayjs(faker.date.recent(40)).format(backendDatoTidformat),
        status: navfaker.random.arrayElement([LestStatus.IkkeLest, LestStatus.Lest]),
        opprettetDato: dayjs(faker.date.recent(40)).format(backendDatoTidformat),
        ferdigstiltDato: dayjs(faker.date.recent(40)).format(backendDatoTidformat),
        erFerdigstiltUtenSvar: ferdigstilUtenSvar,
        ferdigstiltUtenSvarDato: ferdigstilUtenSvar
            ? dayjs(faker.date.recent(40)).format(backendDatoTidformat)
            : undefined,
        ferdigstiltUtenSvarAv: ferdigstilUtenSvar ? getSaksbehandler() : undefined,
        kontorsperretAv: visKontrosperre ? getSaksbehandler() : undefined,
        kontorsperretEnhet: visKontrosperre ? faker.company.companyName() : undefined,
        sendtTilSladding: sladdingNiva !== 0,
        markertSomFeilsendtAv: visMarkertSomFeilsendt ? getSaksbehandler() : undefined,
        erDokumentMelding: meldingstype === Meldingstype.DOKUMENT_VARSEL
    };
}

function sensorerEnkeltOrd(tekst: string): string {
    if (tekst.trim().length === 0) {
        return tekst;
    }
    const ord = tekst.split(' ');
    const sensorOrd = navfaker.random.arrayElement(ord.filter((it) => it.length > 2));
    const sensorering = '*'.repeat(sensorOrd.length);
    return ord.map((it) => (it === sensorOrd ? sensorering : it)).join(' ');
}

function getSaksbehandler(): Saksbehandler {
    return {
        ident: faker.random.alphaNumeric(6),
        fornavn: faker.name.firstName(),
        etternavn: faker.name.lastName()
    };
}

function getMockAutoFullførMap(): AutofullforMap {
    return {
        'bruker.fnr': '10108000398',
        'bruker.fornavn': 'Aremark',
        'bruker.etternavn': 'Testfamilien',
        'bruker.navn': 'Aremark Testfamilien',
        'bruker.subjekt': 'han',
        'bruker.objekt': 'ham',
        'bruker.navkontor': 'NAV Norge',
        'saksbehandler.fornavn': 'Kari',
        'saksbehandler.etternavn': 'Etternavn',
        'saksbehandler.navn': 'Kari Veileder',
        'saksbehandler.enhet': 'NAV Testmark'
    };
}
