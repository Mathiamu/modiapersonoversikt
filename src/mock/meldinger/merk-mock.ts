import faker from 'faker/locale/nb_NO';

export function mockTilgangTilSlett(): Boolean {
    return faker.random.boolean();
}
