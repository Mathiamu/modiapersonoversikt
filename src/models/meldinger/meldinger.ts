import { JournalforingsSak } from '../../app/personside/infotabs/meldinger/traadvisning/verktoylinje/journalforing/JournalforingPanel';
import { Temagruppe } from '../temagrupper';

export interface Traad {
    traadId: string;
    meldinger: Melding[];
}

export interface Melding {
    id: string;
    oppgaveId?: string;
    meldingstype: Meldingstype;
    temagruppe: Temagruppe | null;
    skrevetAvTekst: string;
    journalfortAv?: Saksbehandler;
    journalfortDato?: string;
    journalfortTema?: string;
    journalfortTemanavn?: string;
    journalfortSaksid?: string;
    fritekst: string;
    lestDato?: string;
    status: LestStatus;
    statusTekst?: string;
    opprettetDato: string;
    ferdigstiltDato?: string;
    erFerdigstiltUtenSvar: boolean;
    ferdigstiltUtenSvarAv?: Saksbehandler;
    kontorsperretEnhet?: string;
    kontorsperretAv?: Saksbehandler;
    markertSomFeilsendtAv?: Saksbehandler;
    erDokumentMelding: boolean;
    ferdigstiltUtenSvarDato?: string;
}

export interface Saksbehandler {
    fornavn: string;
    etternavn: string;
    ident?: string;
}

export enum Meldingstype {
    DOKUMENT_VARSEL = 'DOKUMENT_VARSEL',
    OPPGAVE_VARSEL = 'OPPGAVE_VARSEL',
    SPORSMAL_SKRIFTLIG = 'SPORSMAL_SKRIFTLIG',
    SPORSMAL_SKRIFTLIG_DIREKTE = 'SPORSMAL_SKRIFTLIG_DIREKTE',
    SVAR_SKRIFTLIG = 'SVAR_SKRIFTLIG',
    SVAR_OPPMOTE = 'SVAR_OPPMOTE',
    SVAR_TELEFON = 'SVAR_TELEFON',
    DELVIS_SVAR_SKRIFTLIG = 'DELVIS_SVAR_SKRIFTLIG',
    SAMTALEREFERAT_OPPMOTE = 'SAMTALEREFERAT_OPPMOTE',
    SAMTALEREFERAT_TELEFON = 'SAMTALEREFERAT_TELEFON',
    SPORSMAL_MODIA_UTGAAENDE = 'SPORSMAL_MODIA_UTGAAENDE',
    SVAR_SBL_INNGAAENDE = 'SVAR_SBL_INNGAAENDE',
    INFOMELDING_MODIA_UTGAAENDE = 'INFOMELDING_MODIA_UTGAAENDE'
}

export enum LestStatus {
    IkkeLest = 'IKKE_LEST_AV_BRUKER',
    Lest = 'LEST_AV_BRUKER',
    IkkeBesvart = 'IKKE_BESVART'
}

export interface SendReferatRequest {
    fritekst: string;
    temagruppe: Temagruppe;
    meldingstype: Meldingstype.SAMTALEREFERAT_TELEFON | Meldingstype.SAMTALEREFERAT_OPPMOTE;
}

export interface SendSpørsmålRequest {
    fritekst: string;
    sak: JournalforingsSak;
    erOppgaveTilknyttetAnsatt: boolean;
}

export interface SendInfomeldingRequest {
    fritekst: string;
    sak: JournalforingsSak;
}

export interface ForsettDialogRequest {
    traadId: string;
    behandlingsId: string;
    fritekst: string;
    sak?: JournalforingsSak;
    erOppgaveTilknyttetAnsatt: boolean;
    meldingstype: Meldingstype;
    oppgaveId?: string;
}

export interface SendDelsvarRequest {
    traadId: string;
    behandlingsId: string;
    fritekst: string;
    temagruppe: Temagruppe;
    oppgaveId: string;
}

export interface OpprettHenvendelseRequest {
    traadId: string;
}

export interface OpprettHenvendelseResponse {
    behandlingsId: string;
    oppgaveId?: string;
}
