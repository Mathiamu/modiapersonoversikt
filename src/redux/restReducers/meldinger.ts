import { createRestResourceReducerAndActions } from '../../rest/utils/restResource';
import { apiBaseUri } from '../../api/config';
import { AppState } from '../reducers';
import { Traad } from '../../models/meldinger/meldinger';

export function getMeldingerFetchUri(state: AppState): string {
    const fnr = state.gjeldendeBruker.f√∏dselsnummer;
    return `${apiBaseUri}/meldinger/${fnr}/traader`;
}

const { reducer } = createRestResourceReducerAndActions<Traad[]>('meldinger', getMeldingerFetchUri);

export default reducer;
