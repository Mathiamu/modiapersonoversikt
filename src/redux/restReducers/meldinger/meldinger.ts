import { createRestResourceReducerAndActions } from '../../../rest/utils/restResource';
import { apiBaseUri } from '../../../api/config';
import { AppState } from '../../reducers';
import { Traad } from '../../../models/meldinger/meldinger';

function getMeldingerFetchUri(state: AppState): string {
    const fnr = state.gjeldendeBruker.f√∏dselsnummer;
    const enhet = state.session.valgtEnhetId;
    const header = enhet ? `?enhet=${enhet}` : '';
    return `${apiBaseUri}/dialog/${fnr}/meldinger${header}`;
}

export default createRestResourceReducerAndActions<Traad[]>('meldinger', getMeldingerFetchUri);
