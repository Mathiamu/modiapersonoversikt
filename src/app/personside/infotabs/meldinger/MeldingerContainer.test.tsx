import * as React from 'react';
import * as renderer from 'react-test-renderer';
import TestProvider from '../../../../test/Testprovider';
import MeldingerContainer from './MeldingerContainer';
import { getTestStore } from '../../../../test/testStore';
import { statiskTraadMock } from '../../../../mock/meldinger/statiskTraadMock';

test('Viser meldingercontainer med alt innhold', () => {
    const store = getTestStore();
    store.dispatch(store.getState().restResources.tr√•derOgMeldinger.actions.setData([statiskTraadMock]));
    const container = renderer.create(
        <TestProvider customStore={store}>
            <MeldingerContainer />
        </TestProvider>
    );

    const json = container.toJSON();
    expect(json).toMatchSnapshot();
});
