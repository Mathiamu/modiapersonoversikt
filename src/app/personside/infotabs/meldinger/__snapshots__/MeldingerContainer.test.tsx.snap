// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Viser meldingercontainer med alt innhold 1`] = `
.c12 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-animation: fYGRJE 0.3s ease-out;
  animation: fYGRJE 0.3s ease-out;
}

.c12 > * {
  -webkit-flex: 0 1 25rem;
  -ms-flex: 0 1 25rem;
  flex: 0 1 25rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c5 {
  padding: .8rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c5:hover {
  box-shadow: inset 0 0 0 0.1rem #0067c5,0 0.1rem 0.1rem rgba(0,0,0,0.2);
}

.c5:hover > button:last-child {
  padding-left: 0.5rem;
}

.c5 > *:first-child {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c11 {
  border: none;
  padding: 0;
  height: 2rem;
  width: 2rem;
  border-radius: .35rem;
  cursor: pointer;
  background-color: transparent;
  -webkit-transition: padding-left 0.3s;
  transition: padding-left 0.3s;
}

.c11:focus {
  outline: none;
  box-shadow: 0 0 0 0.1875rem #254b6d;
}

.c7 {
  position: relative;
}

.c7 svg {
  height: 1.875rem;
  width: 1.875rem;
  -webkit-clip-path: polygon(0 0,79% 0,79% 46%,100% 46%,100% 100%,0 100%);
  clip-path: polygon(0 0,79% 0,79% 46%,100% 46%,100% 100%,0 100%);
}

.c8 {
  position: absolute;
  top: -0.1875rem;
  right: 0.5625rem;
  border-radius: 50%;
  padding: 0 0.1rem;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c9 .order-first {
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
}

.c9 .order-second {
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c6 > *:first-child {
  padding-right: .8rem;
}

.c4:focus {
  outline: none;
  box-shadow: 0 0 0 0.1875rem #254b6d;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  margin-top: 0.2rem;
}

.c10 > *:not(:last-child) {
  margin-bottom: 0.2rem;
  margin-right: 0.2rem;
}

.c2 {
  padding: .8rem;
}

.c1 {
  border-radius: 0.25rem;
  background-color: white;
  box-shadow: 0 0 0.5rem rgba(0,0,0,0.1),0 0 0 1px rgba(0,0,0,0.1);
}

.c1 ol {
  list-style: none;
}

.c3 > *:not(:first-child) {
  border-top: solid 0.0625rem #b7b1a9;
}

.c0 {
  position: relative;
}

.c0 > * {
  margin-bottom: .8rem;
}

.c0 > *:first-child {
  -webkit-flex: 30% 1 1;
  -ms-flex: 30% 1 1;
  flex: 30% 1 1;
}

.c0 > *:last-child {
  -webkit-flex: 70% 1 1;
  -ms-flex: 70% 1 1;
  flex: 70% 1 1;
}

@media (min-width:62.5rem) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-align-items: flex-start;
    -webkit-box-align: flex-start;
    -ms-flex-align: flex-start;
    align-items: flex-start;
    max-height: 100%;
  }

  .c0 > *:last-child {
    margin-left: .8rem;
  }

  .c0 > * {
    overflow-y: auto;
    max-height: 100%;
  }
}

<article
  className="c0"
>
  <nav
    className="c1"
  >
    <div
      className="c2"
    >
      <div
        className="skjemaelement move-input-label"
      >
        <label
          className="skjemaelement__label"
          htmlFor="Helt tilfeldig ID"
        >
          SÃ¸k etter melding
        </label>
        <input
          className="skjemaelement__input input--fullbredde"
          id="Helt tilfeldig ID"
          type="text"
        />
        <div
          aria-live="assertive"
          role="alert"
        />
      </div>
    </div>
    <ol
      className="c3"
    >
      <li
        className="test-meldinger-melding c4"
        tabIndex={-1}
      >
        <div
          className="c5"
          onClick={[Function]}
        >
          <div>
            <div
              className="c6"
            >
              <span
                className="c7"
              >
                <svg
                  height={24}
                  viewBox="0 0 24 24"
                  width={24}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g
                    fill="none"
                    stroke="#000"
                    strokeLinejoin="round"
                    strokeMiterlimit="10"
                  >
                    <path
                      d="M7.44 14.564c-.489-.066-.971-.17-1.44-.314l-5 2 2-3.5c-1.573-1.259-2.5-3.125-2.5-5.064 0-3.83 3.806-6.936 8.5-6.936s8.5 3.106 8.5 6.936"
                    />
                    <path
                      d="M9.5 16.25c0 4.289 5 7.5 9.5 5.5l4 1.5-1.5-3c1.296-1.037 2-2.402 2-4 0-3.154-3.134-6-7-6s-7 2.846-7 6z"
                    />
                  </g>
                </svg>
                <p
                  className="typo-undertekst-bold c8"
                >
                  4
                </p>
              </span>
              <div>
                <div
                  className="c9"
                >
                  <p
                    className="typo-element order-second"
                  >
                    Delsvar - Arbeid
                  </p>
                  <p
                    className="typo-normal order-first"
                  >
                    13.08.2019 00:00
                  </p>
                </div>
                <div
                  className="c10"
                >
                  <div
                    className="etikett etikett--fokus"
                  >
                    <span
                      className="typo-normal"
                    >
                      Under arbeid
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <button
            aria-label="Vis meldinger for Delsvar - Arbeid"
            aria-selected={false}
            className="c11"
          >
            <i
              className="nav-frontend-chevron chevronboks chevron--stor chevron--hoyre"
            />
          </button>
        </div>
      </li>
    </ol>
  </nav>
  <div>
    <div
      className="c12"
    />
  </div>
</article>
`;
